assistive_gym/envs/human_comfort_env.py

```diff
@@ -70,6 +74,86 @@ def _get_obs(self, agent=None): # not needed
     def get_total_force(self):
         pass
 
+    
+    def get_reba_score(self):
+        human_dict = HumanUrdfDict()
+        rebaScore = RebaScore()
+        # list joints in the order required for a reba score
+        joints = ["head", "neck", "left_shoulder", "left_elbow", "left_lowarm", "right_shoulder", "right_elbow", "right_lowarm", 
+            "left_hip", "left_knee", "left_ankle", "right_hip", "right_knee", "right_ankle", "left_hand", "right_hand"]
+        
+        # obtain the links in the right order for the rebascore code
+        dammy_ids = []
+        for joint in joints:
+            dammy_ids.append(human_dict.get_dammy_joint_id(joint))
+
+        # use dammy ids to obtain the right link, use the right knee as the root joint
+        pose = []
+        root = p.getLinkState(self.human.body, dammy_ids[12])[0] # root joint
+
+        for i in dammy_ids:
+            # get the location of each dammy joint and append to the pose list
+            loc = p.getLinkState(self.human.body, i)[0]
+            norm_loc = [loc[0] - root[0], loc[1] - root[1], loc[2] - root[2]]
+            pose.append(norm_loc)
+        
+        # convert to numpy array
+        pose = np.array(pose)
+        # print("pose: ", pose)
+
+        # print("angles: ", angles, "\nlength: ", len(angles))
+        sample_pose = np.array([[ 0.08533354,  1.03611605,  0.09013124],
+                      [ 0.15391247,  0.91162637, -0.00353906],
+                      [ 0.22379057,  0.87361878,  0.11541229],
+                      [ 0.4084777 ,  0.69462843,  0.1775224 ],
+                      [ 0.31665226,  0.46389668,  0.16556387],
+                      [ 0.1239769 ,  0.82994377, -0.11715403],
+                      [ 0.08302169,  0.58146328, -0.19830338],
+                      [-0.06767788,  0.53928527, -0.00511249],
+                      [ 0.11368726,  0.49372503,  0.21275574],
+                      [ 0.069179  ,  0.07140968,  0.26841402],
+                      [ 0.10831762, -0.36339359,  0.34032449],
+                      [ 0.11368726,  0.41275504, -0.01171348],
+                      [ 0.        ,  0.        ,  0.        ],
+                      [ 0.02535541, -0.43954643,  0.04373671],
+                      [ 0.26709431,  0.33643749,  0.17985192],
+                      [-0.15117603,  0.49462711,  0.02703403]])
+
+        # following code is from the ergonomic repo (https://github.com/rs9000/ergonomics/blob/master/ergonomics/reba.py)
+        body_params = rebaScore.get_body_angles_from_pose_right(pose)
+        arms_params = rebaScore.get_arms_angles_from_pose_right(pose)
```

Comment:
I'm thinking here they are using the joint angle to calculate, not the position. we are feeding joint cardesian position in the pose array, arent we?

Reply:
ohhh I just checked their code. I might be wrong though. Seems like we need the position
https://github.com/rs9000/ergonomics/blob/master/ergonomics/reba.py#L269

But since they are converting inside their function to cal the angle from position array, I am not sure if the angle they calculated vs our joint angle we set to the model is actually align. Could you try to print out those 2 angles to make sure those are align? Any small error (for ex: passing in wrong order/ passing center of mass position instead of end position) might incur in error